### PR TITLE
Automatic restart skip

### PR DESCRIPTION
**Pull request by bena-nasa**
If a field makes it to ExtData, go through the hierarchy and make any component with that field in it's import MAPL_RestartSkip. Added new routine in MAPL_Generic to do this. This was requested by the chemistry folks so they don't have to remember to add restartskip and if something may or may not be provided by ExtData depending on the configuration of the system.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [X] I have tested this change with a run of GEOSgcm (if non-trivial)
- [X] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [X] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Review by atrayano — APPROVED** [review]

File: generic/MAPL_Generic.F90 [hunk 1]
```diff
@@ -11334,4 +11339,43 @@ subroutine warn_empty(string, MPL, rc)
       _RETURN(ESMF_SUCCESS)
    end subroutine warn_empty
 
+   recursive subroutine MAPL_AddI4AttributeToFields(gc,field_name,att_name,att_val,rc)
```
Comment (tclune) [inline generic/MAPL_Generic.F90:11342]:
```suggestion
   recursive subroutine MAPL_AddAttributeToFields_I4
   (gc,field_name,att_name,att_val,rc)
```
Nominal preference to put the disambguators at the end to make them easier to spot.  I think this is more consistent with other cases, though I'm sure it is far from uniform in MAPL.  No need to change if this is the only issue I find.

File: generic/MAPL_Generic.F90 [hunk 2]
```diff
@@ -11334,4 +11339,43 @@ subroutine warn_empty(string, MPL, rc)
...
+         end if
+         call ESMF_AttributeSet(field,name=att_name,value=att_val,_RC)
+      end if
+      nc = state%get_num_children()
+      if (nc > 0) then
+         do i=1,nc
+            child_gc => state%get_child_gridcomp(i)
+            call MAPL_AddI4AttributeToFields(child_gc,field_name,att_name,att_val,_RC)
+         enddo
+      end if
```
Comment (tclune) [inline generic/MAPL_Generic.F90:11376]: Conditional is redundant.
```suggestion
      do i=1,nc
         child_gc => state%get_child_gridcomp(i)
         call MAPL_AddI4AttributeToFields(child_gc,field_name,att_name,att_val,_RC)
      enddo
```

File: gridcomps/Cap/MAPL_CapGridComp.F90 [hunk 3]
```diff
@@ -690,8 +690,9 @@ subroutine initialize_history(cap, rc)
   end subroutine initialize_history
 
 
-  subroutine initialize_extdata(cap , rc)
+  subroutine initialize_extdata(cap , root_gc, rc)
```
Comment (tclune) [inline gridcomps/Cap/MAPL_CapGridComp.F90:693]: This will definitely give me some thought about how to treat in MAPL 3

**Review by tclune — CHANGES_REQUESTED** [review]
Does any of this require an update as ESMF_Info on the V3 branch?

Minor changes requested.

**Comment by mathomp4** [comment]
> Does any of this require an update as ESMF_Info on the V3 branch?

@tclune Yes, it will. The ESMF_AttributeGet call will need to be an infoget. 

Probably a good time to do a sweep of MAPL3 to see if any ESMF_Attribute calls have snuck in. (Maybe there is a way at ESMF_init time to pass a flag that doesn't allow any ESMF_Attribute calls? That way we can be strict.)

File: generic/MAPL_Generic.F90 (hunk 1 above)
Comment (bena-nasa) [reply generic/MAPL_Generic.F90]: Made the change

**Review by tclune — APPROVED** [review]

**Pull request merged**